src/auth/use-check-if-account-creation-finished.ts

```diff
@@ -1,30 +1,36 @@
 import {Dispatch, useEffect} from 'react';
-import {FirebaseAuthTypes} from '@react-native-firebase/auth';
+import auth from '@react-native-firebase/auth';
 import {AuthReducerAction, AuthStatus} from './types';
+import Bugsnag from '@bugsnag/react-native';
 
 /*
  * Check whether the user creation is finished up to 10 times with an interval
  * of 1 second.
  */
 export const useCheckIfAccountCreationFinished = (
-  user: FirebaseAuthTypes.User | null,
+  userId: string | undefined,
   authStatus: AuthStatus,
   dispatch: Dispatch<AuthReducerAction>,
 ) => {
   useEffect(() => {
-    if (user && authStatus === 'creating-account') {
+    if (userId && authStatus === 'creating-account') {
+      const user = auth().currentUser;
       let retryCount = 0;
       const intervalId = setInterval(async () => {
```

Comment:
Should probably look into replacing this setInterval as well

Reply:
Yeah. Won't do it right now as it shouldn't be a much risk since it is 1sec between each run, and capped at 10 times.